### PR TITLE
Move `worker_pool_service.get_worker` to `worker_service.get`

### PR DESCRIPTION
**Pull request by kiendang**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: packages/syft/src/syft/service/worker/worker_service.py [hunk 1]
```diff
@@ -160,17 +160,19 @@ def list(self, context: AuthedServiceContext) -> Union[SyftSuccess, SyftError]:
...
-                    message=f"Failed to update status for worker: {worker.id}. Error: {result.err()}"
+        with contextlib.closing(docker.from_env()) as client:
+            for idx, worker in enumerate(workers):
+                worker_ = _check_and_update_status_for_worker(
+                    client=client,
+                    worker=worker,
+                    worker_stash=self.stash,
+                    credentials=context.credentials,
                 )
-            workers[idx] = worker
```
Comment (kiendang) [inline packages/syft/src/syft/service/worker/worker_service.py:173]: @shubham3121 There was a bug here. Earlier this was
```py
for idx, worker in enumerate(workers):
    result = check_and_update_status_for_worker(
    ...
workers[idx] = worker
```

`workers[idx]` was not updated correctly, still using the one with outdated status.

Comment (shubham3121) [reply packages/syft/src/syft/service/worker/worker_service.py:173]: Good catch 👍🏾 . Thanks for the fix.

File: packages/syft/src/syft/service/worker/worker_service.py [hunk 2]
```diff
@@ -182,37 +184,46 @@ def status(
...
+        self, context: AuthedServiceContext, uid: UID
+    ) -> Union[SyftWorker, SyftError]:
         result = self.stash.get_by_uid(credentials=context.credentials, uid=uid)
         if result.is_err():
             return SyftError(message=f"Failed to retrieve worker with UID {uid}")
-        worker: SyftWorker = result.ok()
 
-        if context.node.in_memory_workers:
-            return worker.status, worker.healthcheck
+        worker = cast(SyftWorker, result.ok())
```
Comment (shubham3121) [inline packages/syft/src/syft/service/worker/worker_service.py]: Can we handle another condition here, if `result.ok()` is None then return `SyftError(message=f"Worker doesn't exists for id: {uid}")`

**Review by shubham3121 — CHANGES_REQUESTED** [review]

File: packages/syft/src/syft/service/worker/worker_service.py (hunk 2 above)
Comment (shubham3121) [reply packages/syft/src/syft/service/worker/worker_service.py]: Resolve by 01c89f3

**Review by shubham3121 — APPROVED** [review]
Nice work @kiendang !! 💯

File: packages/syft/src/syft/store/document_store.py [hunk 3]
```diff
@@ -699,7 +699,7 @@ def update(
         credentials: SyftVerifyKey,
         obj: BaseStash.object_type,
         has_permission=False,
-    ) -> Optional[Result[BaseStash.object_type, str]]:
+    ) -> Result[BaseStash.object_type, str]:
```
Comment (kiendang) [inline packages/syft/src/syft/store/document_store.py:702]: @shubham3121 Fixed the type here. This just returns a `Result` not an `Optional[Result]`.

**Pull request merged**